### PR TITLE
[OSDOCS-7645]: Remove Note Regarding Inability to Delete Default Machine Pool, other MP updates 

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.13+
Issue:

Link to docs preview:

https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#creating_a_machine_pool_rosa-managing-worker-nodes

https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#deleting-machine-poolsrosa-managing-worker-nodes

https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#rosa-adding-taints-clirosa-managing-worker-nodes

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by bchandra-ocp** [comment]
https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about#machine-pools - 
1. We need to update the list to reflect a) You can remove or add labels to any machine pool b) You can add taint to any machine pool as long as there is one machine pool without any taints c) You can create or delete a machine pool as long as there is one machine pool without any taints.
2. "By default, a cluster has one machine pool. During cluster installation, you can label this machine pool. After the cluster is installed its node type or size cannot be changed." This needs to be corrected: "By default, a cluster has one machine pool. During cluster installation, you can define instance type or size and add labels to this machine pool."
3. The NOTE in the box already covers for the fact that once created, machine pool can't be updated to change the node type or size. 

https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about#machine-pools-in-multiple-zone-clusters

The whole section starting with "When you create a machine pool..." can be simplified into following two sentences that help customer with the feature:

1. In a cluster created across multiple Availability Zones (AZ), the machine pools can be created across either all the three AZs or any single AZ of your choice. The machine pool created by default at the time of cluster creation will be created with machines in all 3 AZs and scale in multiples of 3. 
 
https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes
1. What's the difference between two bullet points under 5 (b) "Set the minimum and maximum node count limits for autoscaling."? Perhaps, we need to say what are allowed minimums and allowed values (multiples of 3 in case of 3 AZs)? 
2. The NOTE about optionally configuring autoscaling is already covered by saying step 5 is Optional. Do we need that NOTE box? 
3. Step 7 to Optionally configure disk size must be expanded. It should include what's default, what's minimum and maximum values etc.
4. Step 8: same comment about NOTE box. We can remove that since the step is Optional already
5. Step 8 c: Needs a NOTE to call out that creating a machine pool with Taint will be possible only if the cluster already has a machine pool w/o taint. 
6.

File: modules/creating-a-machine-pool-ocm.adoc [hunk 1]
```diff
@@ -86,6 +86,10 @@ endif::openshift-rosa[]
 .. Expand the *Edit node labels and taints* menu.
 .. Under *Node labels*, add *Key* and *Value* entries for your node labels.
 .. Under *Taints*, add *Key* and *Value* entries for your taints.
+[NOTE]
+====
+Creating a machine pool with taints is only possible if the cluster already has at least one machine pool without a taint.
```
Comment (xueli181114) [inline modules/creating-a-machine-pool-ocm.adoc:91]: Hmm... In theory a cluster will always to have a machinepool without taint. So customer can create machinepool with taint anytime.

File: modules/rosa-adding-taints-cli.adoc [hunk 2]
```diff
@@ -10,6 +10,10 @@
 
 You can add taints to a machine pool for your Red Hat OpenShift Service on AWS (ROSA) cluster by using the ROSA CLI.
 
+[NOTE]
+====
+For users of ROSA CLI `rosa` version 1.2.25 and prior versions, the number of taints cannot be changed within the machine pool (ID=`Default`) created along with the cluster. cannot be deleted. For users of ROSA CLI `rosa` version 1.2.26 and beyond, the number of taints can be changed within the machine pool (ID=`worker`) created along with the cluster.
```
Comment (xueli181114) [inline modules/rosa-adding-taints-cli.adoc:15]:
```suggestion
For users of ROSA CLI `rosa` version 1.2.25 and prior versions, the number of taints cannot be changed within the machine pool (ID=`Default`) created along with the cluster cannot be deleted. For users of ROSA CLI `rosa` version 1.2.26 and beyond, the number of taints can be changed within the machine pool (ID=`worker`) created along with the cluster.
```

File: rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc [hunk 3]
```diff
@@ -27,24 +27,26 @@ Machine pools are a higher level construct to compute machine sets.
...
 
-By default, a cluster has one machine pool. During cluster installation, you can label this machine pool. After the cluster is installed, the `Default` machine pool cannot be deleted, and its node type or size cannot be changed.
+By default, a cluster has one machine pool. During cluster installation, you can define instance type or size and add labels to this machine pool.
 
 After a cluster's installation:
 
-* You can relabel a `Default` machine pool.
+* You can remove or add labels to any machine pool.
 * You can add additional machine pools to an existing cluster.
+* You can add taints to any machine pool as long as there is one machine pool without any taints.
```
Comment (xueli181114) [inline rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc:36]: The machinepool should at the same time have untaint 2(for single zone) and 3(for multi zone) replicas
```
lixue@Xue-Lis-MacBook-Pro ~ % rosa edit machinepool worker -c sdq-ci-twppd --taints kwy=value:NoSchedule
E: Failed to update machine pool 'worker' on cluster 'sdq-ci-twppd': Machine pool 'worker' for cluster '263hpcg0hqb5v4qkp94taut6rnn4apat' can't be updated with provided params. At least one machine pool able to run OCP workload is required. Pool should not have taints and minimum replica count should be '2'
lixue@Xue-Lis-MacBook-Pro ~ % rosa list machinepools -c sdq-ci-twppd                                   
ID      AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS                                                                TAINTS    AVAILABILITY ZONES                    SUBNETS                                                                         SPOT INSTANCES  DISK SIZE  
sz1     No           1         m5.xlarge                                                                                      us-west-2b                            subnet-02c45e21191037c15                                                        No              300 GiB
worker  Yes          3-6       m5.xlarge      label1/openshift.com=labelvalue1, label2/openshift.com=labelvalue2              us-west-2a, us-west-2b, us-west-2c    subnet-0bc1a548e9a8b46e1, subnet-02c45e21191037c15, subnet-09416be08e29bd88c    No              150 GiB
```

File: rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc [hunk 4]
```diff
@@ -27,24 +27,26 @@ Machine pools are a higher level construct to compute machine sets.
...
+* You can create or delete a machine pool as long as there is one machine pool without any taints.
```
Comment (xueli181114) [inline rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc:37]: Same with above

**Comment by xueli181114** [comment]
Any preview changes for OSD part?
Should we clarify about non-ccs cluster cannot delete/taints the default worker pool in the doc? Needs PM's confirmation

**Comment by xueli181114** [comment]
And per the preview file https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#rosa-managing-worker-nodes
It shows `Worker node instance type` but on UI it's `Compute node instance type`
And so do `Worker node count` should be `Compute node count (per zone)`

File: modules/creating-a-machine-pool-ocm.adoc [hunk 5]
```diff
@@ -86,6 +86,10 @@ endif::openshift-rosa[]
...
+====
 .. For each taint, select an *Effect* from the drop-down menu. Available options include `NoSchedule`, `PreferNoSchedule`, and `NoExecute`.
```
Comment (xueli181114) [inline modules/creating-a-machine-pool-ocm.adoc:93]: Above highlight message make this line wrapped as another charactered list which starts with `a` again.

**Comment by xueli181114** [comment]
And per tuning config part
https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#rosa-adding-tuning_rosa-managing-worker-nodes
tuning config only work for hosted cluster, it's not highlighted
And commands output should be updated, there is no Default worker pool for hosted cluster.

**Comment by mletalie** [comment]
> https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about#machine-pools -
> 
> 1. We need to update the list to reflect a) You can remove or add labels to any machine pool b) You can add taint to any machine pool as long as there is one machine pool without any taints c) You can create or delete a machine pool as long as there is one machine pool without any taints.
> 2. "By default, a cluster has one machine pool. During cluster installation, you can label this machine pool. After the cluster is installed its node type or size cannot be changed." This needs to be corrected: "By default, a cluster has one machine pool. During cluster installation, you can define instance type or size and add labels to this machine pool."
> 3. The NOTE in the box already covers for the fact that once created, machine pool can't be updated to change the node type or size.
> 
> https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about#machine-pools-in-multiple-zone-clusters
> 
> The whole section starting with "When you create a machine pool..." can be simplified into following two sentences that help customer with the feature:
> 
> 1. In a cluster created across multiple Availability Zones (AZ), the machine pools can be created across either all the three AZs or any single AZ of your choice. The machine pool created by default at the time of cluster creation will be created with machines in all 3 AZs and scale in multiples of 3.
> 
> https://64189--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes
> 
> 1. What's the difference between two bullet points under 5 (b) "Set the minimum and maximum node count limits for autoscaling."? Perhaps, we need to say what are allowed minimums and allowed values (multiples of 3 in case of 3 AZs)?
> 2. The NOTE about optionally configuring autoscaling is already covered by saying step 5 is Optional. Do we need that NOTE box?
> 3. Step 7 to Optionally configure disk size must be expanded. It should include what's default, what's minimum and maximum values etc.
> 4. Step 8: same comment about NOTE box. We can remove that since the step is Optional already
> 5. Step 8 c: Needs a NOTE to call out that creating a machine pool with Taint will be possible only if the cluster already has a machine pool w/o taint.

Addressed everything in this comment except #1, #2 and #3 in last section of comment. Outside the scope of the ticket, and this needs to get merged by 9/11/23. Will address these concerns in another ticket asap.

File: modules/creating-a-machine-pool.adoc [hunk 6]
```diff
@@ -6,4 +6,8 @@
 [id="creating_a_machine_pool_{context}"]
 = Creating a machine pool
 
-A default machine pool is created when you install a {product-title} (ROSA) cluster. After installation, you can create additional machine pools for your cluster by using {cluster-manager} or the ROSA CLI (`rosa`).
+A machine pool is created when you install a {product-title} (ROSA) cluster. After installation, you can create additional machine pools for your cluster by using {cluster-manager} or the ROSA CLI (`rosa`).
+[NOTE]
+====
+For users of ROSA CLI `rosa` version 1.2.25 and prior versions, the machine pool created along with the cluster is identified as `Default`. For users of ROSA CLI  `rosa` version 1.2.26 and beyond, the machine pool created along with the cluster is identified as `worker`.
```
Comment (tmalove) [inline modules/creating-a-machine-pool.adoc:12]:
```suggestion
For users of ROSA CLI `rosa` version 1.2.25 and earlier versions, the machine pool created along with the cluster is identified as `Default`. For users of ROSA CLI  `rosa` version 1.2.26 and later, the machine pool created along with the cluster is identified as `worker`.
```

File: modules/deleting-machine-pools-cli.adoc [hunk 7]
```diff
@@ -7,6 +7,11 @@
 = Deleting a machine pool using the ROSA CLI
 You can delete a machine pool for your Red Hat OpenShift Service on AWS (ROSA) cluster by using the ROSA CLI.
 
+[NOTE]
+====
+For users of ROSA CLI `rosa` version 1.2.25 and prior versions, the machine pool (ID='Default') created along with the cluster cannot be deleted. For users of ROSA CLI `rosa` version 1.2.26 and beyond, the machine pool (ID='worker') created along with the cluster can be deleted as long as there is one machine pool within the cluster that contains no taints and at least two replicas for a single-AZ cluster or three replicas for a multi-AZ cluster.
```
Comment (tmalove) [inline modules/deleting-machine-pools-cli.adoc:12]:
```suggestion
For users of ROSA CLI `rosa` version 1.2.25 and earlier versions, the machine pool (ID='Default') that is created along with the cluster cannot be deleted. For users of ROSA CLI `rosa` version 1.2.26 and later, the machine pool (ID='worker') that is created along with the cluster can be deleted as long as there is one machine pool within the cluster that contains no taints, and at least two replicas for a single-AZ cluster or three replicas for a multi-AZ cluster.
```
Does 'AZ' modify 'cluster'? Should it be 'single AZ-cluster' and 'multi AZ-cluster'?

File: modules/deleting-machine-pools.adoc [hunk 8]
```diff
@@ -16,21 +16,13 @@ endif::openshift-rosa[]
...
-//
-
 ifndef::openshift-rosa[]
+
 .Prerequisites
 
 * You have created an {product-title} cluster.
 * The cluster is in the ready state.
-* You have an existing machine pool without any taints and with at least two instances for a single-AZ cluster or three instances for a multi-AZ cluster.
+* You have an existing machine pool without any taints and with at least two replicas for a single-AZ cluster or three replicas for a multi-AZ cluster.
```
Comment (tmalove) [inline modules/deleting-machine-pools.adoc:25]: See my earlier question about 'AZ-cluster'.

File: modules/rosa-adding-taints-cli.adoc [hunk 9]
```diff
@@ -10,6 +10,10 @@
 
 You can add taints to a machine pool for your Red Hat OpenShift Service on AWS (ROSA) cluster by using the ROSA CLI.
 
+[NOTE]
+====
+For users of ROSA CLI `rosa` version 1.2.25 and prior versions, the number of taints cannot be changed within the machine pool (ID=`Default`) created along with the cluster. For users of ROSA CLI `rosa` version 1.2.26 and beyond, the number of taints can be changed within the machine pool (ID=`worker`) created along with the cluster. There must be at least one machine pool without any taints and with at least two replicas for a single-AZ cluster or three replicas for a multi-AZ cluster.
```
Comment (tmalove) [inline modules/rosa-adding-taints-cli.adoc:15]:
```suggestion
For users of ROSA CLI `rosa` version 1.2.25 and earlier versions, the number of taints cannot be changed within the machine pool (ID=`Default`) that are created along with the cluster. For users of ROSA CLI `rosa` version 1.2.26 and later, the number of taints can be changed within the machine pool (ID=`worker`) that are created along with the cluster. There must be at least one machine pool without any taints and with at least two replicas for a single-AZ cluster or three replicas for a multi-AZ cluster.
```

File: rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc [hunk 10]
```diff
@@ -27,24 +27,26 @@ Machine pools are a higher level construct to compute machine sets.
...
 ====
 * You can add a label to each added machine pool.
-* You cannot delete the `Default` machine pool. However, you can delete the non-default machine pools. 
 
 Multiple machine pools can exist on a single cluster, and each machine pool can contain a unique node type and node size configuration.
 
 == Machine pools in multiple zone clusters
-When you create a machine pool in a multiple availability zone (Multi-AZ) cluster, that one machine pool has 3 zones. The machine pool, in turn, creates a total of 3 compute machine sets - one for each zone in the cluster. Each of those compute machine sets manages one or more machines in its respective availability zone.
+In a cluster created across multiple Availability Zones (AZ), the machine pools can be created across either all of the three AZs or any single AZ of your choice. The machine pool created by default at the time of cluster creation will be created with machines in all three AZs and scale in multiples of three.
```
Comment (tmalove) [inline rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc:48]:
```suggestion
In a cluster that is created across multiple Availability Zones (AZ), the machine pools can be created across either all of the three AZs or any single AZ of your choice. The machine pool created by default at the time of cluster creation will be created with machines in all three AZs and scale in multiples of three.
```

**Review by tmalove — APPROVED** [review]

**Pull request merged**